### ISSUE TITLE
[BUG]:flutter sound is depreciated

### ISSUE THREAD

**Issue by mkassir95** [Canardoux/flutter_sound#807]
I am using  Flutter Sound LITE  Version  8.4.2 , and I am getting this warning  : 
Note: C:\Users\user\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_sound_lite-8.4.2\android\src\main\java\com\dooboolab\fluttersound\FlutterSound.java uses or overrides a deprecated API.

**Comment by alexda12** [comment]
This plugin is being retired : #780

**Comment by mkassir95** [comment]
what do  you mean , there will not be any updated versions?

**Comment by alexda12** [comment]
@mkassir95 No updates for 45 days to this plugin and based on comments in #780 the project will shortly be terminated.

**Comment by mkassir95** [comment]
he did not says this , he say other people will continue working on the project

**Comment by Larpoux** [comment]
No @alexda12 ,
I never said that this project will be retired.
I just said that if nobody work on a project, it will certainly die slowly.

It is same for all Open Source Projects. Everything depend on developers wanting to contribute to it.
I am not the only one developer on earth.
I have worked on Flutter Sound during two years. But I do not want to spend all the rest of my life on this project.

Recently Flutter Sound was forked to two branches : 
- Flutter Sound 8.x (MPL license)
- Tau Sound 9.0 (GPL license)

For myself, I prefer not to work anymore on a MPL project. But of course other developers can do that: I will continue to merge Pull Requests if any, for sure.
I will continue to work on Tau Sound 9.0, but not full time anymore. I want also to work on other projects, also.
But other developers are welcome to work on Tau 9.0.
@Alvarocda for example is currently working on a rewrite of this f...ing UI Widgets.
**We really need other developers.**

**Comment by alexda12** [comment]
@Larpoux @mkassir95 Ok, maybe I misunderstood what was written here where it states **you have decided to terminate Tau 9.x in a few weeks** 

but I agree - there is no point working on a MPL project as your simply wasting your time - If I take on TAU development , will I have complete autonomy to do what I like ? (I dont like the idea of bundling all of the codecs with the full version of the plugin and also forcing the things you have done with the manifest re: background audio service)

> 
> I had stopped working on Tau for several weeks.
> But I just decide now to try to terminate Tau 9.0. Probably in a few weeks ...
> 
> But it is very important that other developers will work on Tau soon.
> I will not be available for tau in the long run.

**Comment by Larpoux** [comment]
My english is really bad. I am sorry.
I was trying to say that Tau 9.0 has been actually in a beta state for a long time. Too long !
I really want to release it (specially the API documentation, which is a hard job for me).
So I write that I need to terminate the release of 9.0.
I did not want to say that I will terminate completely working on Tau

**Comment by Larpoux** [comment]
> I dont like the idea of bundling all of the codecs with the full version of the plugin and also forcing the things you have done with the manifest re: background audio service

Yes, I agree with you and with @sallypeters : better to handle audio service with another plugin (like `justAudioService`).
Actually I have not yet decided if I will completely remove the code from Tau, or only in the LITE flavor.
I must look to `justAudioService` to see if it is not too much complicate to use it for a casual developer.

**Comment by alexda12** [comment]
I have looked at **justAudioService** the documentation is very detailed and there are **full examples** that work seamlessly. The API is very straightforward even for **novice developers** so I think since all of the heavy lifting has already been done with this self contained plugin , we should remove the service-based code in Flutter Sound (Or maybe we should move forward with TAU 10.x now ?) I think it doesnt make sense to maintain all of the sound plugins , i.e FS, FSL, TAU - there are just far too many codebases and especially when you also factor in different licences GPL, MPL etc

I think in view of the amount of work that you personally have done on this project it would be a pity to let this project die ... 

What help do you need to get the TAU 9.0 release out ?

**Comment by Larpoux** [comment]
Hi, @alexda12 ,

> I think in view of the amount of work that you personally have done on this project it would be a pity to let this project die ...

When I read this, it makes me feel good. This is much more positive than saying that we must running away elsewhere because this project is  discontinued.
My problem is that I feel very lonely when I work on τ. Everybody want improvements and fixes but nobody help.
And I am never sure that my work is useful.
And sometime I am completely discouraged and I cannot see a computer without wanting to vomit.

> What help do you need to get the TAU 9.0 release out ?

There are not many things to do. The most important thing is to write the new API documentation.
I am probably the only person who can do that.

I want also to write a page explaining how to use `justAudioService` inside a τ App. And write a small example.
Your feedback on using this plugin is very interesting. I like very much the idea to remove everything about **Now Playing**.
For a stupid reason : the code to support Audio Service inside τ has been written by @salvatore373 . But he does not maintain his code, and I have never spend time to master this part of Flutter Sound.

There is one thing where someone can be very helpful : 
@Alvarocda rewrote the τ UI Widgets. But he has not enough time to finish the documentation, the examples, and the tests.
Someone should review his code, and give him feedback.

**Comment by alexda12** [comment]
@Larpoux Sorry for the delay in getting back to you , I was ill with Covid but now recovered ...

Regarding : 

> I want also to write a page explaining how to use justAudioService inside a τ App. And write a small example.
> Your feedback on using this plugin is very interesting. I like very much the idea to remove everything about Now Playing.
> For a stupid reason : the code to support Audio Service inside τ has been written by @salvatore373 . But he does not maintain his code, and I have never spent time to master this part of Flutter Sound.

I have implemented The Audio Service but since Flutter Sound also has a service we obviously get issues and the app gets terminated (You need to remove your audio service  bindings so TAU/Flutter sound coexists with Just Audio's Service) . 

I can take ownership of creating a sample app with Flutter Sound (Or  TAU if you prefer to use that instead) and Audio Service if you like and also write a page documenting this so we can add it to your support page  ... Let me know if this will help bring back some life into Flutter Sound/TAU. 

Finally - since you took over Flutter Sound, an incredible amount of work has been done on your part - I think we should continue with this project , so lets decide the direction and what is achievable for 2022 and then set a roadmap so existing/new users can see what the plans are and hence there will be more excitement/enthusiasm with this project.

**Comment by Larpoux** [comment]
Hi @alexda12 ,
I am sorry to hear that you got this terrible virus.
Myself I very often have some health problems : I fall in a great depression and cannot see a computer without wanting to vomit.
When I was young (50 years ago!) I was certain that I could do everything by myself, and did not accept to use code from someone else.
But now I am old, and I am not happy to code alone. 
There are two kinds of free project : 
- Some project are done by a single developer. Some of them are true genius.
- Other projects are done by a community working together

I prefer the later, now. I am not a genius and I have many things to learn from other developers.
And I think that a very important force with Open Source come from many developers each doing just simple things.
The number of developers is important.

Now, I am going to re-start working on τ. I was not sure to want continue working on this project alone during 2022 year.
But when I read your mail I feel much stronger.



> can take ownership of creating a sample app with Flutter Sound (Or TAU if you prefer to use that instead) and Audio Service if you like and also write a page documenting this so we can add it to your support page ... Let me know if this will help bring back some life into Flutter Sound/TAU.

# OH, YES. IT WILL DEFINITELY HELP BRING BACK SOME LIFE INTO τ

By the way, what do you think better : remove the audio service both on the LITE and FULL flavor, or just in the LITE flavor ?
My current thinking is that if it is easy to use Just Audio Service,  then better to remove the audio-service everywhere.
It will simplify the API, the documentation, the code, and certainly remove some bugs.

Perhaps we can continue talking together on [Discord](https://discord.gg/MWNfvaTHmQ)
